pkg/services/search/service_test.go

```diff
@@ -0,0 +1,57 @@
+package search
+
+import (
+	"testing"
+
+	"github.com/grafana/grafana/pkg/bus"
+	"github.com/grafana/grafana/pkg/models"
+	m "github.com/grafana/grafana/pkg/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSearch_SortedResults(t *testing.T) {
+	bus.AddHandler("test", func(query *FindPersistedDashboardsQuery) error {
+		query.Result = HitList{
+			&Hit{Id: 16, Title: "CCAA", Type: "dash-db", Tags: []string{"BB", "AA"}},
+			&Hit{Id: 10, Title: "AABB", Type: "dash-db", Tags: []string{"CC", "AA"}},
+			&Hit{Id: 15, Title: "BBAA", Type: "dash-db", Tags: []string{"EE", "AA", "BB"}},
+			&Hit{Id: 25, Title: "bbAAa", Type: "dash-db", Tags: []string{"EE", "AA", "BB"}},
+			&Hit{Id: 17, Title: "FOLDER", Type: "dash-folder"},
+		}
+		return nil
+	})
+
+	bus.AddHandler("test", func(query *m.GetUserStarsQuery) error {
+		query.Result = map[int64]bool{10: true, 12: true}
+		return nil
+	})
+
+	bus.AddHandler("test", func(query *m.GetSignedInUserQuery) error {
+		query.Result = &m.SignedInUser{IsGrafanaAdmin: true}
+		return nil
+	})
+
+	svc := &SearchService{}
+
+	query := &Query{
+		Limit: 2000,
+		SignedInUser: &models.SignedInUser{
+			IsGrafanaAdmin: true,
+		},
+	}
+
+	err := svc.searchHandler(query)
+	assert.Nil(t, err)
```

Comment:
I think you should use `require.Nil` here, since otherwise the test will continue after a failure in `svc.searchHandler`, which doesn't really make sense and leads to chaotic output.